cli/azd/pkg/commands/builder.go

```diff
@@ -29,7 +31,17 @@ func Build(action Action, rootOptions *GlobalCommandOptions, use string, short s
 			azCli := GetAzCliFromContext(ctx)
 			ctx = context.WithValue(ctx, environment.AzdCliContextKey, azCli)
 
-			return action.Run(ctx, cmd, args, azdCtx)
+			// This is done to simply mock behavior. We could either get the full command invocation path
+			// using GetCommandPath, or more than likely, ask for the event name as a Builder argument
```

Comment:
The idea is that we'll update our builder to be `Build(action Action, rootOptions *GlobalCommandOptions, use string, short string, long string, telemetryEventName string) *cobra.Command`.

If `telemetryEventName` is set, then we'll record the action span and log the event.

Currently, not all commands use the Builder, and we'll need a change to make it consistent.

Reply:
Rephrasing the question: Should `command.Build` in builder be our entry point for defining whether telemetry should be emitted for a particular command? I think it makes sense, but we'll need to refactor all commands to use it and to avoid direct construction of `cobra.Command` types in the future.